third_party/terraform/resources/resource_dataflow_job.go

```diff
@@ -275,17 +275,17 @@ func resourceDataflowJobDelete(d *schema.ResourceData, meta interface{}) error {
 
 		_, updateErr := resourceDataflowJobUpdateJob(config, project, region, id, job)
 		if updateErr != nil {
-			gerr, isGoogleErr := err.(*googleapi.Error)
```

Comment:
honestly, I feel like we should just use a new terminated check + isRetryableError, i.e. 

```golang
_, updateErr := resourceDataflowJobUpdateJob(config, project, region, id, job)
if isDataflowJobTerminatedError(updateError) {
  // Job has already been terminated, skip.
  return nil
}

if isRetryableError(updateErr, isDataflowErrorNotReadyForCanceling) {
  return resource.RetryableError(updateErr)
}
return resource.NonRetryableError(updateErr)

}

where `isDataflowJobTerminatedError` and `isDataflowErrorNotReadyForCanceling` are helper error handlers/new retry predicate for "not yet ready for canceling", etc). That way we actually check against other errors that aren't googleapi errors (URL/net errors, etc)

The terminated error util should be in this file, but you can add the dataflow one in error_retry_predicates.go or in this file.

Reply:
Alternatively, if we edit this to use retryTimeDuration, we get that by default. The retry function would just include 

```
_, updateErr := resourceDataflowJobUpdateJob(config, project, region, id, job)
if isDataflowJobTerminatedError(updateError) {
  // Job has already been terminated, skip.
  return nil
}
return updateErr
```

and you'd pass the retry predicate to retryTimeDuration

Reply:
Turns out this is a little bit more effort than originally thought, and should be presented in a separate PR.